include/bit_algorithm_details.hpp

```diff
@@ -140,21 +140,22 @@ ForwardIt word_shift_right(ForwardIt first,
                           typename ForwardIt::difference_type n
 )
 {
+    using word_type = typename ForwardIt::value_type;
     auto d = distance(first, last);
     if (n <= 0 || n >= d) return first;
-    ForwardIt last2 = first;
-    ForwardIt d_first = first;
-    std::advance(last2, distance - n - 1);
-    std::advance(d_first, n);
-    std::copy(first, last2, d_first);
-    for (; first != d_first; ++first) {
-        *first = 0;
-    }
-    return first;
+    auto it = first;
+    std::advance(it, d-n);
+    std::rotate(first, it, last);
+    it = first;
+    std::advance(it, n);
+    std::fill(first, it, 0); 
```

Comment:
I'm having issues compiling this function for single word iterators. The errors seem to point at this `std::fill` call

Reply:
```
TestType x = 99;
bit::bit_iterator<TestType*> f(&x, 2);
bit::bit_iterator<TestType*> l(&x, 5);
bit::word_shift_right(f, l, 2);
```

Reply:
Ok, I guess this function was not intended to take bit_iterators, so never mind.

Reply:
Hmm, while it _is_ only intended for words, I feel like it should be able to work w/ bit_iterators as well... However I'm not terribly concerned as this will end up being replaced by the std version. What do you think?

Reply:
fine by me